### DISCUSSION TITLE
Custom imagemagick commands & disabling higher resolution images from accessed through url

### DISCUSSION THREAD

**Discussion by strivn** [alshedivat/al-folio#2899] [Q&A]
Hi! So my questions are: 

1. Can we add additional imagemagick commands to be used? I was thinking of adding stripping metadata and some others
2. Can we limit access to e.g. assets/image.png (the original image) from url/assets/image.png and instead redirect / refer them to for example url/assets/image-1400.webp instead?

Thank you!

**Comment by george-gca** [comment]
1. We currently use [jekyll-imagemagick](https://gitlab.com/emmmile/jekyll-imagemagick) plugin to handle image conversion with imagemagick. You can check its configurations and see if it fits your needs.
2. Take a look at the output of the build under the `_site/` directory. See how the build handles, for example, the profile page. I believe this is what you are talking about?

**Comment by strivn** [comment]
Hi, thanks for the swift response!

1. Ah nvm, I think it was due to some caching that my newer configurations aren't being reflected in the output. After I deleted the generated webp images, it worked as it should. Is there a way to not use cache? 
2. Let me reclarify: Since all images under `assets/img` are generated into `_site`, this'd include the original image. I prefer to remove/exclude the original image from being built into `_site`. Is there a way to do this? 

Thank you!

**Reply by george-gca** [reply]
1. What you mean about cache? In GitHub pages, or locally? Independently, every time you build your site all the contents in `_site/` get overwritten, so this should not be a problem.
2. Not that I am aware of, maybe there is some [jekyll-imagemagick](https://gitlab.com/emmmile/jekyll-imagemagick) config that allows this. But I think this is intended. Taking from its docs:

---

### How to use WEBP in HTML

Not all browsers are compatible with WEBP. Using the `<picture>` element and specifying all image formats available is the best option. This way the browser will decide which format to use based on its own capabilities.

```html
<picture>
  <source srcset="/path/to/image.webp" type="image/webp">
  <img src="/path/to/image.jpg" alt="">
</picture>
```
